Compatibility with phpDocumentor Reflection Dockbloc 3

dunglas opened the pull request:
Follow up of #264. Allow to use Prophecy (and PHPUnit) with Symfony 3.1.

Test that it works with @fabiang's patch on ReflectionDocBlock merged: https://github.com/phpDocumentor/ReflectionDocBlock/pull/72

The last commit (update of  `composer.json` to use the branch of Fabian) must be removed before the merge in Prophecy.

ping @mvriel

GrahamCampbell commented on composer.json:
```diff
@@ -16,9 +16,16 @@
+    "repositories": [
+        {
+            "type": "vcs",
+            "url": "https://github.com/fabiang/ReflectionDocBlock"
```
This can't be merged as is, because composer doesn't inherit this from packages that want to install prophecy.

dunglas replied:
Right. It was just to run a Travis build, it will be removed before merge (see https://github.com/phpspec/prophecy/pull/275#issuecomment-222499112).

dunglas commented:
Travis failure is normal. We only need to get phpDocumentor/ReflectionDocBlock#72 and Prophecy will be compatible with Symfony 3.1.

stof commented on src/Prophecy/PhpDocumentor/MethodTagRetrieverInterface.php:
```diff
@@ -0,0 +1,28 @@
+use phpDocumentor\Reflection\DocBlock\Tags\Method;
+
+/**
+ * @author Théo FIDRY <theo.fidry@gmail.com>
```
please mark this interface and all its implementations as ``@internal``

dunglas replied:
done

Soullivaneuh commented on .travis.yml:
```diff
@@ -14,6 +14,16 @@ branches:
+  fast_finish: true
+  include:
+    - php: '7.0'
+      env: PHPDOCUMENTOR_REFLECTION_DOCBLOCK="~2.0"
```
Maybe take profit to use the new constraint format: `^2.0`?

dunglas commented:
A fixed version of Reflection Docblock has been released: https://github.com/phpDocumentor/ReflectionDocBlock/pull/72#issuecomment-223879914

I've updated this PR to use it and I fixed the last remaining comments. It should be ready to be merged now.

stof commented:
I will do a release this evening

dunglas commented:
Thanks @stof
